docs/kubernetes_setup.md

```diff
@@ -0,0 +1,188 @@
+        ports:
+        - name: serverport
+          containerPort: 8131
+          hostPort: 8131
+        - name: agentport
+          containerPort: 8132
+          hostPort: 8132
+        - name: adminport
+          containerPort: 8133
+          hostPort: 8133
+        volumeMounts:
+        - name: konnectivity-server-pki
+          mountPath: /etc/kubernetes/pki/
+          readOnly: true
+      volumes:
+      - name: konnectivity-server-pki
+        hostPath:
+          # the path on your master where the certificate are located
+          path: /var/lib/konnectivity-server/pki/
+```
+
+The `apiserver-network-proxy` server should now be running on the master.
+
+## Nodes setup
+
+We will use the following DaemonSet in order to deploy the `apiserver-network-proxy` agent on the worker nodes:
+
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    k8s-app: konnectivity-agent
+  namespace: kube-system
+  name: konnectivity-agent
+spec:
+  selector:
+    matchLabels:
+      k8s-app: konnectivity-agent
+  template:
+    metadata:
+      labels:
+        k8s-app: konnectivity-agent
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+    spec:
+      priorityClassName: system-cluster-critical
+      tolerations:
+        - key: "CriticalAddonsOnly"
+          operator: "Exists"
+      hostNetwork: true
+      volumes:
+        - name: pki
+          hostPath:
+            # the path of your certificates on the worker nodes
+            path: /var/lib/konnectivity-agent/pki
+      containers:
+        - image: gcr.io/google-containers/proxy-agent:v0.0.3
+          name: konnectivity-agent
+          command: 
+          - /proxy-agent
+          args: 
+          - --logtostderr=true
+          - --ca-cert=/etc/srv/kubernetes/pki/konnectivity-agent/ca.crt
+          - --agent-cert=/etc/srv/kubernetes/pki/konnectivity-agent/client.crt
+          - --agent-key=/etc/srv/kubernetes/pki/konnectivity-agent/client.key
+          - --proxy-server-host=<URL of the API server>
+          - --proxy-server-port=8132
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+          resources:
+            limits:
+              cpu: 50m
+              memory: 30Mi
+          livenessProbe:
+            httpGet:
+              host: 127.0.0.1
+              port: 8093
+              path: /healthz
+            initialDelaySeconds: 15
+            timeoutSeconds: 15
+          volumeMounts:
+            - name: pki
+              mountPath: /etc/srv/kubernetes/pki/konnectivity-agent
+```
+
+## Verification
+
+In order to verify that the setup is working, you can try to exec a `kubectl logs`, and check in the `apiserver-network-proxy` agent and server that packets are indeed being forwarded.
```

Comment:
I would also add something about looking in the konnectivity server/agent logs to ensure it actually went through this egress. (If ssh tunnel is enabled on the KAS it actually goes through that egress. Not too worried as that option is deprecated and will be going away)